File: commands/moderation/ban.js
module.exports = {
    name: 'ban',
    permissions: ["BAN_MEMBERS"],
    cooldown: 0,
    execute(client, message, cmd, args, Discord) {

        const target = message.mentions.users.first();
        const reason = args.join(' ')

        if (target) {
            const memberTarget = message.guild.members.cache.get(target.id);
            memberTarget.ban();
<<<<<<< HEAD
            message.channel.send({ embed: { color: `#00f2ff`, description: `${memberTarget} have been banned! Reason: *****${reason}*****` } })
=======
            message.channel.send({ embed: { color: `RANDOM`, description: `${memberTarget} have been banned! Reason: *****${reason}*****` } })
>>>>>>> jsmerge

            const image = message.guild.iconURL({ dynamic: true })
            const userID = message.mentions.users.first()
            const personbaned = client.users.cache.get(userID.id)
            const baned = new Discord.MessageEmbed()
<<<<<<< HEAD
                .setColor('#00f2ff')
=======
                .setColor('RANDOM')
>>>>>>> jsmerge
                .setTitle('You have been Banned!')
                .addField('By', message.author.toString(), true)
                .addField('Server', message.guild.name, true)
                .addField('Reason', reason)
                .setThumbnail(image)
                .setTimestamp();

            personbaned.send(baned);
        } else {
<<<<<<< HEAD
            message.channel.send({ embed: { color: `#00f2ff`, description: 'Please enter the name whome to Ban' } })
=======
            message.channel.send({ embed: { color: `RANDOM`, description: 'Please enter the name whome to Ban' } })
>>>>>>> jsmerge
        }
    }
}

File: commands/Moderation/ban.py
from setting import *

@commands.has_permissions(ban_members=True)
@client.command()
async def ban(ctx, member : discord.Member, *, reason: str="No reason"):
    
    # Banning member
    await member.ban(reason=reason)
    await ctx.send(embed=discord.Embed(title="banned a user", description=f"{member.mention} has been banned for `{reason}`", color=0x00f2ff))

# Error(s)
@ban.error
async def ban_error(ctx, error):
    
    # Missing Argument(s)
    if isinstance(error, commands.MissingRequiredArgument):
        embed=discord.Embed(title="Missing Argument(s)", description='Please mention the user to ban', color=0x00f2ff)
        embed.add_field(name="Example", value=f"{prefix}ban {ctx.author.mention}")
        await ctx.send(embed=embed)
        
    # Member not found
    elif isinstance(error, commands.MemberNotFound):
        await ctx.send(embed=discord.Embed(title="Member not found", description="Member you mentioned is not in the server!", color=0x00f2ff))
    
    # Role not high enough
    elif isinstance(error, commands.CommandInvokeError):
        await ctx.send(embed=discord.Embed(title="Role not high enough", description=f"This user is at the top of the role hierarchy then me. Please put me on the top in order to perform this task!", color=0x00f2ff))
    
    # Missing permision
    elif isinstance(error, commands.MissingPermissions):
        await ctx.send(embed=discord.Embed(title="Missing Permision", description=f"Your dont have permission to ban, Noob", color=0x00f2ff))
    
    # Unknown error
    else:
        await unknown_error(ctx, error)